numpy/fft/tests/test_pocketfft.py

```diff
@@ -156,6 +156,14 @@ def test_all_1d_norm_preserving(self):
                     assert_array_almost_equal(x_norm,
                                               np.linalg.norm(tmp))
 
+    def test_dtypes(self):
+        # make sure that all input precisions are accepted and internally
+        # converted to 64bit
+        for c in "fdg":
+            x = random(30).astype(np.dtype(c))
+            assert_array_almost_equal(np.fft.ifft(np.fft.fft(x)), x)
+            assert_array_almost_equal(np.fft.irfft(np.fft.rfft(x)), x)
+
```

Comment:
I'm not using the "FDG" types because that would mean calling a real-valued FFT with a complex-valued input, which really makes no sense (and it produces a warning as expected).
I can add "e", if you like.

Reply:
I'm sorry, I don't have the time right now to learn how `pytest` works.

Reply:
```
@pytest.mark.parametrize("dtype", [np.half, np.single, np.double, np.longdouble])
def test_something(dtype):
   ...
```

Reply:
Thanks!